.github/workflows/publish-commit-bottles.yml

```diff
@@ -68,14 +69,17 @@ jobs:
         with:
           signing_key: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY }}
 
+      - name: Checkout PR branch
+        run: gh pr checkout ${{github.event.inputs.pull_request}}
+
       - name: Pull and upload bottles to GitHub Packages
         env:
           BREWTESTBOT_NAME_EMAIL: "BrewTestBot <1589480+BrewTestBot@users.noreply.github.com>"
           HOMEBREW_GPG_PASSPHRASE: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY_PASSPHRASE }}
           HOMEBREW_GITHUB_API_TOKEN: ${{secrets.HOMEBREW_CORE_PUBLIC_REPO_EMAIL_TOKEN}}
           HOMEBREW_GITHUB_PACKAGES_USER: brewtestbot
           HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{secrets.HOMEBREW_CORE_GITHUB_PACKAGES_TOKEN}}
-        run: brew pr-pull --debug --workflows=tests.yml --committer="$BREWTESTBOT_NAME_EMAIL" --root-url="https://ghcr.io/v2/homebrew/core" ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
+        run: brew pr-pull --debug --workflows=tests.yml --committer="$BREWTESTBOT_NAME_EMAIL" --root-url="https://ghcr.io/v2/homebrew/core" --no-autosquash --clean ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
```

Comment:
@Bo98 Alternatively, we can (and should) care less about commit message/format pedantry, particularly if/when we're moving to merge commits where it matters much less about narrowing things down to a single commit.

Reply:
That would be disappointing - `git blame` is actually really useful here, compared to sometimes being horrid to use on Homebrew/brew. But I'll consider the rest of the concept first.

Reply:
> I'm also not entirely sure how `brew extract`/`FormulaVersions` behaves when a version bump is split into several commits but that can be adjusted if need be.

Yea, I think we may break `brew extract` and `FormulaVersions`. Though I think the former will be broken because it relies on the latter. Breaking `FormulaVersions` would not be good -- we use it for some audits.